src/index.ts

```diff
@@ -1,101 +1,128 @@
+      });
+
+      break;
+  }
+};
+
+const handleAnyEvent = async (context: SlackContext) => {
+  const { type } = context.event._rawEvent;
+
+  // Handle bot mentions (probably someone wants to send some doge)
+  if (type === MessageTypes.MENTION) {
+    const message = context.event._rawEvent.text;
+    const tipPattern = /(<@[\w]*>)(\s*)(tip)(\s*)((<@[\w]*>\s?)*)(\s*)(\d*)/gi;
+    const randomPattern = /(<@[\w]*>)(\s*)(random)(\s*)(\d*)/gi;
+    const rainPattern = /(<@[\w]*>)(\s*)(rain)(\s*)(\d*)/gi;
+
+    const tipMatches = [...message.matchAll(tipPattern)];
+    const randomMatches = [...message.matchAll(randomPattern)];
+    const rainMatches = [...message.matchAll(rainPattern)];
+
+    // TIP
+    if (tipMatches.length > 0) {
+      const amount = Number(tipMatches[0][8]);
+      const recipientsPart = tipMatches[0][5];
+      const recipientMatches = [...recipientsPart.matchAll(/(@[\w]*)/g)];
+      const recipients = recipientMatches.map((item) =>
+        item[0].replace('@', '')
+      );
+
+      if (amount < 2) {
+        await context.chat.postEphemeral({
+          text: `Come on, don't be stingy! I can't even pay the network rent with that!.`,
+        });
+        return;
+      }
+      await sendDoge(context, blockIo, recipients, amount);
+      return;
+    }
 
-bot.onEvent(async (context: Context) => {
-  let command;
-  let commandText;
+    // RANDOM
+    if (randomMatches.length > 0) {
+      const amount = Number(randomMatches[0][5]);
+      const members: string[] = await getChannelMembers(context);
+      const pickedMember = getRandomArrayElements(members, 1);
+      await sendDoge(context, blockIo, pickedMember, amount);
+      return;
+    }
 
-  // Public / Private channels
-  if (context.event.isChannelsMessage || context.event.isGroupsMessage) {
-    // Unless @cooldoge is mentioned, don't react
-    if (
-      context.event.text &&
-      !context.event.text.includes(`<@${process.env.BOT_USER_ID}>`)
-    ) {
+    // RAIN
+    if (rainMatches.length > 0) {
+      const maxMembers = 5;
+      const amount = rainMatches[0][5];
+      const members = await getChannelMembers(context);
+      const pickedMembers: string[] = getRandomArrayElements(
+        members,
+        maxMembers
+      );
+      await sendDoge(
+        context,
+        blockIo,
+        pickedMembers,
+        amount / pickedMembers.length
+      );
       return;
     }
 
-    commandText = context.event.text
-      .trim()
-      .split(' ')[1]
-      .toLowerCase();
-
-    command = pick(commands, ['tip', 'rain', 'random', 'help'])[commandText];
-
-    // DM with @cooldoge
-  } else if (context.event.isText) {
-    commandText = context.event.text
-      .trim()
-      .split(' ')[0]
-      .toLowerCase();
-    command = pick(commands, ['balance', 'deposit', 'withdraw', 'help'])[
-      commandText
-    ];
-  } else {
-    return;
+    // Command mismatched. So, send help message to the user
+    help(context);
```

Comment:
fyi this help is not sent in DM even if I type `help`

Reply:
Done. ;)